packages/react-core/src/components/OptionsMenu/OptionsMenuToggleWithText.tsx

```diff
@@ -51,27 +52,78 @@ export const OptionsMenuToggleWithText: React.FunctionComponent<OptionsMenuToggl
   /* eslint-enable @typescript-eslint/no-unused-vars */
   'aria-label': ariaLabel = 'Options menu',
   ...props
-}: OptionsMenuToggleWithTextProps) => (
-  <div
-    className={css(
-      styles.optionsMenuToggle,
-      styles.modifiers.text,
-      isPlain && styles.modifiers.plain,
-      isDisabled && styles.modifiers.disabled,
-      isActive && styles.modifiers.active
-    )}
-    {...props}
-  >
-    <span className={css(styles.optionsMenuToggleText, toggleTextClassName)}>{toggleText}</span>
-    <button
-      className={css(styles.optionsMenuToggleButton, toggleButtonContentsClassName)}
-      id={`${parentId}-toggle`}
-      aria-haspopup="listbox"
-      aria-label={ariaLabel}
-      aria-expanded={isOpen}
-      onClick={() => onToggle(!isOpen)}
+}: OptionsMenuToggleWithTextProps) => {
+  const buttonRef = React.useRef<HTMLButtonElement>();
+
+  React.useEffect(() => {
+    document.addEventListener('mousedown', onDocClick);
+    document.addEventListener('touchstart', onDocClick);
+    document.addEventListener('keydown', onEscPress);
+    return () => {
+      document.removeEventListener('mousedown', onDocClick);
+      document.removeEventListener('touchstart', onDocClick);
+      document.removeEventListener('keydown', onEscPress);
+    };
+  });
```

Comment:
Thank you for your suggestions. However, there are some issues when changing the  `document` to `parentRef`.

1. When the dropdown menu is open, you press `Tab`, it will not focus back on the dropdown button but the next button.
2. When the dropdown menu is open, you click on the other dropdown buttons in other examples on this doc, this dropdown menu will not hide.

I think it may because there are other `eventListeners` that set in `Toggle.tsx` file under Dropdown Component, which will affect the behavior of this button. If I use `parentRef` instead of `document`, the behavior of this Dropdown button might not be consistent with the other Dropdown buttons on this doc. However, if I use `document` here, it can keep in accordance with the other buttons.